plugin-server/src/worker/plugins/run.ts

```diff
@@ -33,11 +35,52 @@ async function runSingleTeamPluginOnEvent(
     const onEventPayload = convertToOnEventPayload(event)
 
     try {
+        // Check if we should record HTTP calls
+        const recordHttpCalls = hub.DESTINATION_MIGRATION_DIFFING_ENABLED === true && hub.TASKS_PER_WORKER === 1
+
+        if (recordHttpCalls) {
+            // Clear the HTTP call recorder before running the plugin
+            getHttpCallRecorder().clearCalls()
```

Comment:
Hmmmmm. This check means we will only clear and log if the settings are enabled. But if we aren't clearing then isn't the recorder just going to build a huge list of requests in memory that never clears?